lib/WTSI/DNAP/Warehouse/Schema/Query/LibraryDigest.pm

```diff
@@ -497,11 +500,19 @@ sub _add_entity {
 }
 
 sub _accept {
```

Comment:
How about reworking this so that it only takes the ```IseqProductMetric``` row and gets the ```IseqFlowcell``` itself?
It's always called in the same way isn't it? e.g.
```
if (!$self->_accept($fc_row) | !$self->_accept($prow)) {
```
(the bitwise or ```|``` instead of ```||``` or ```or``` seems odd to me also).
So how about moving that or logic inside the function, perhaps get rid of the negatives (```_accept``` becomes ```_skip```) as well:
```
if ($self->_skip($prow)) {
```
and then change the internals to do a
```
 if ($quality_field ne 'qc') {$row=$row->iseq_flowcell}
```
Not looked at this before so I may have wrong end of stick....